### PR TITLE
[AIRFLOW-4792] Include Sentry into Contrib Airflow

### PR DESCRIPTION
**Pull request by tiopi**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title:
https://issues.apache.org/jira/browse/AIRFLOW-4792

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Added Sentry into the contrib folder of Airflow. This hook allows easy connection to the Sentry SDK for specific DAGs.

Differences between #5416 and #5407:
#5416:  PR focuses on adding Sentry functionality to specific DAGs. This is achieved by adding a sentry hook to the contrib folder. Users can then simply call the hook with their `SENTRY_DSN` for each DAG.
#5407: PR focuses on adding Sentry to the Airflow ecosystem. Its aim is to handle all errors within airflow (Scheduler, worker, DAGs, tasks).  (This is the continuation of #5382, which I closed by accident.)

### Tests

- [x] My PR adds the following unit tests to `sentry_hook.py`.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by mik-laj** [comment]
This is your third PR with the integration of Sentry. Can you summarize what are the differences between them?
https://github.com/apache/airflow/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Atiopi+
It is difficult to talk and work out the best solution when there are several places to discuss

**Comment by tiopi** [comment]
> This is your third PR with the integration of Sentry. Can you summarize what are the differences between them?
> https://github.com/apache/airflow/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Atiopi+
> It is difficult to talk and work out the best solution when there are several places to discuss

I updated the description on both of my open PRs to reflect the differences between the each addition.

File: airflow/contrib/hooks/sentry_hook.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        else:
+            import logging
+            from sentry_sdk.integrations.logging import LoggingIntegration
+
+            sentry_logging = LoggingIntegration(
+                level=logging.INFO, event_level=logging.ERROR
+            )
+            integrations = [sentry_logging]
+
+        conn = self.get_connection(self.sentry_conn_id)
```
Comment (mik-laj) [inline airflow/contrib/hooks/sentry_hook.py:67]: Why do you think it is important to keep it in the database? This will cause a huge load. The static value in the configuration file seems to be a better idea. Holding data in a database allows you to store data in a more secure way but Sentry DSN is safe.

Comment (kyle-hamlin) [reply airflow/contrib/hooks/sentry_hook.py:67]: +1 on having DNS in config

File: scripts/ci/pylint_todo.txt [hunk 2]
```diff
@@ -107,6 +107,7 @@
 ./airflow/contrib/hooks/sagemaker_hook.py
 ./airflow/contrib/hooks/salesforce_hook.py
 ./airflow/contrib/hooks/segment_hook.py
+./airflow/contrib/hooks/sentry_hook.py
```
Comment (mik-laj) [inline scripts/ci/pylint_todo.txt:110]: New files should be fixed and not ignored.  This list contains only files that are not compatible when we introduce the pylint. The files from this list are gradually deleted.
